package.json

```diff
@@ -53,29 +53,34 @@
     "eslint-plugin-react": "^6.4.1",
     "eslint-plugin-standard": "^3.0.1",
     "jest": "^19.0.2",
-    "less": "^2.7.2",
-    "less-loader": "^4.0.5",
-    "patternfly": "^3.27.4",
+    "node-sass": "^4.7.2",
+    "patternfly": "^3.31.0",
     "prettier": "^1.7.4",
     "raf": "^3.4.0",
     "react": "^16.0.0",
     "react-dom": "^16.0.0",
     "react-test-renderer": "^16.0.0",
+    "sass-loader": "^6.0.6",
     "semantic-release": "^6.3.6",
     "style-loader": "^0.19.0"
   },
   "peerDependencies": {
-    "patternfly": "^3.27.4",
+    "patternfly": "^3.31.0",
     "prop-types": "^15.5.10",
     "react": "^15.6.2 || ^16.0.0",
     "react-dom": "^15.6.2 || ^16.0.0"
   },
+  "sassIncludes": {
+    "patternfly": "--include-path node_modules/patternfly/dist/sass",
+    "bootstrap": "--include-path node_modules/patternfly/node_modules/bootstrap-sass/assets/stylesheets",
+    "fontAwesome": "--include-path node_modules/patternfly/node_modules/font-awesome-sass/assets/stylesheets"
+  },
   "scripts": {
     "start": "npm run storybook:run",
     "commit": "git-cz",
-    "build": "npm run build:scripts && npm run build:less",
-    "build:scripts": "babel src --out-dir lib --ignore test.js,__mocks__",
-    "build:less": "mkdir -p lib/styles/less && cp -r less/ lib/styles/less",
+    "build": "npm run build:scripts && npm run build:sass",
+    "build:scripts": "babel src --out-dir dist/js --ignore test*.js,__mocks__",
+    "build:sass": "mkdir -p dist/sass && cp -r sass/patternfly-react/* dist/sass && node-sass --output-style compressed --include-path sass $npm_package_sassIncludes_patternfly $npm_package_sassIncludes_bootstrap $npm_package_sassIncludes_fontAwesome -o dist/css sass/patternfly-react.scss",
```

Comment:
Do I understand it right that compiled sass file which patternfly-react distributes includes also patternfly base styles? I strongly believe that we want to avoid this. Doing this completely removes ability to customize patternfly styles/variables by consuming application. It is IMHO very important thing to be able to do. Why not just produce additional custom styles which patternfly-react adds?

Also making a strict cut and switch directly to sass cuts off applications which already uses less. patternfly-react needs to be compatible with a way patternfly is meant to be extended. Which (correct me if I am wrong) is https://github.com/openstack/tripleo-ui/blob/master/src/less/base.less - important thing is that there needs to be a way to override patternfly and bootstrap variables from consuming app.

Ideally, patternfly-react.sass is just another file to add for the consumming app. Also if we're distributing sass files, we should also distribute less to support apps which did not switch yet. To get rid of the problem we could produce compiled css file.

Reply:
No, the patternfly-react css does not include patternfly base styles, just the variables. Those are necessary in order to compile the sass.

We don't currently have any Less files, so I don't feel we need to continue to support it. I don't think we want to follow the path of supporting both less and sass so we need to choose one, I feel Sass is the obvious choice.

Correct adding the partial _patternfly-react.sass will only include the patternfly react component specific sass.